open-hackathon/src/hackathon/expr/expr_mgr.py

```diff
@@ -67,9 +73,12 @@ def __report_expr_status(self, expr):
             vms = db_adapter.find_all_objects_by(UserResource,
                                                  type=VIRTUAL_MACHINE,
                                                  status=RUNNING,
-                                                 user_template_id=expr.user_template.id)
+                                                 template_id=expr.template.id)
             vms_id = map(lambda v: v.id, vms)
-            for vm_config in db_adapter.find_all_objects_by(VMConfig, virtual_machine_id=vms_id):
+            vms_all = []
+            for id in vms_id:
+                vms_all.append(db_adapter.find_first_object_by(VMConfig, virtual_machine_id=id))
+            for vm_config in vms_all:
```

Comment:
also ,please add more comments. I'm worry about that we cannot understand what those codes are doing in future.